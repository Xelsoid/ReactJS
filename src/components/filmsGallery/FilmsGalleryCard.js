import React from 'react';
import PropTypes from "prop-types";

import './filmsGalleryCard.scss';

const FilmsGalleryCard = ({film}) => {
  const {title, poster_path, release_date, genres} = film;

  return (
    <div className="film-gallery-card">
      <img alt={title} className="film-gallery-card__image" src={poster_path} />
      <div className="clearfix">
        <span className="film-gallery-card__name">{title}</span>
        <span className="film-gallery-card__year">{release_date}</span>
      </div>
      <small className="film-gallery-card__genre">{genres.map(genre => `${genre} `)}</small>
    </div>
  )
};

FilmsGalleryCard.defaultProps = {
  film: null
};

FilmsGalleryCard.propTypes = {
  film: PropTypes.instanceOf(Object),
};

export default FilmsGalleryCard;
