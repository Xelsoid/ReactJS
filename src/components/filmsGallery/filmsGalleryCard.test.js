import React from 'react';
import FilmsGalleryCard from './FilmsGalleryCard';
import DATA from "../../mockedData/MOCKED_DATA";

const film = DATA.data[1];

describe('FilmsGalleryCard', () => {
  it('renders correctly', () => {
    const component = shallow(
      <FilmsGalleryCard film={film} />
    );
    expect(component).toMatchSnapshot();
  });
});
