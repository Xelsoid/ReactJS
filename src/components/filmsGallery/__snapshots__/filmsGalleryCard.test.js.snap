// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmsGalleryCard renders correctly 1`] = `
<div
  className="film-gallery-card"
>
  <img
    alt="Zootopia"
    className="film-gallery-card__image"
    src="https://image.tmdb.org/t/p/w500/sM33SANp9z6rXW8Itn7NnG1GOEs.jpg"
  />
  <div
    className="clearfix"
  >
    <span
      className="film-gallery-card__name"
    >
      Zootopia
    </span>
    <span
      className="film-gallery-card__year"
    >
      2016-02-11
    </span>
  </div>
  <small
    className="film-gallery-card__genre"
  >
    Animation 
    Adventure 
    Family 
    Comedy 
  </small>
</div>
`;
